hazelcast/src/main/java/com/hazelcast/core/HazelcastInstance.java

```diff
@@ -452,6 +453,15 @@
      */
     @Nonnull CPSubsystem getCPSubsystem();
 
+    /**
+     * Returns a service to execute distributed SQL queries.
+     *
+     * @return SQL service
+     *
+     * @see SqlService
+     */
+    @Nonnull SqlService getSql();
```

Comment:
Shouldn't we also mark this method as `@Beta`?

Reply:
I think it is sufficient to mark the service itself as `@Beta`. The method `HazelcastInstance.getSql` is not in the beta state on its own. It is the service that it provides is in the beta status. We used this pattern for all other `@Beta` usages.

Reply:
I agree it's technically sufficient, but we're trying to make it as clear as possible for the users. All APIs available through this method are beta, so even though if this method itself is not, there's nothing you can do with it if you want to avoid beta.

I would even document the state of the development here or in the `SqlService` class docs. It will set expectations and reduce disillusionment.

Reply:
Ok, agree. Added `@Beta` annotation to `HazelcastInstance.getSql`. Regarding concrete capabilities - it is nearly impossible to list and maintain the list of supported features. In SQL this is the goal of documentation.

Reply:
I would add a statement like this:
>The SQL module in the 4.1 release is in beta state. We might change details of the behavior in future releases or even change the API. The functionality is also limited, for example we don't yet support joins or aggregations or DML statements. We also don't support IMaps with NATIVE in-memory format (enterprise feature). However, the features that are implemented are of production quality and are safe to be used in production settings.

Reply:
Added a note about beta state. 
As mentioned before, I am not keen to add a description of specific features to JavaDoc. This is a goal of documentation, not JavaDoc, to inform users about supported SQL features. Whatever we write in JavaDoc will be outdated almost immediately. Even the list of unsupported feature in the sample snippet is not accurate already.